Src/EB/AMReX_EB2_2D_C.cpp

```diff
@@ -20,87 +21,96 @@ void set_eb_data (const int i, const int j,
     constexpr Real tiny = 1.e-15;
 #endif
 
-    const Real axm = apx(i  ,j  ,0);
-    const Real axp = apx(i+1,j  ,0);
-    const Real aym = apy(i  ,j  ,0);
-    const Real ayp = apy(i  ,j+1,0);
-    const Real apnorm = std::hypot(axm-axp,aym-ayp) + 1.e-30_rt;
-    const Real nx = (axm-axp) * (1.0_rt/apnorm);
-    const Real ny = (aym-ayp) * (1.0_rt/apnorm);
+    const Real axm = apx(i  ,j  ,0)*dx[1];
+    const Real axp = apx(i+1,j  ,0)*dx[1];
+    const Real aym = apy(i  ,j  ,0)*dx[0];
+    const Real ayp = apy(i  ,j+1,0)*dx[0];
+    const Real daxp = (axm-axp);
+    const Real dayp = (aym-ayp);
+    const Real apnorm = std::hypot(daxp,dayp) + 1.e-30_rt;
+    const Real nx = daxp * (1.0_rt/apnorm);
+    const Real ny = dayp * (1.0_rt/apnorm);
+    const Real bareascaling = std::sqrt( (nx*dx[0])*(nx*dx[0]) +
+            (ny*dx[1])*(ny*dx[1]) );
 
     const Real nxabs = amrex::Math::abs(nx);
     const Real nyabs = amrex::Math::abs(ny);
 
     Real x_ym, x_yp, y_xm, y_xp;
     if (nx > 0.0_rt) {
-        x_ym = -0.5_rt + aym;
-        x_yp = -0.5_rt + ayp;
+        x_ym = -0.5_rt*dx[0] + aym;
+        x_yp = -0.5_rt*dx[0] + ayp;
     } else {
-        x_ym = 0.5_rt - aym;
-        x_yp = 0.5_rt - ayp;
+        x_ym = 0.5_rt*dx[0] - aym;
+        x_yp = 0.5_rt*dx[0] - ayp;
     }
 
     if (ny > 0.0_rt) {
-        y_xm = -0.5_rt + axm;
-        y_xp = -0.5_rt + axp;
+        y_xm = -0.5_rt*dx[1] + axm;
+        y_xp = -0.5_rt*dx[1] + axp;
     } else {
-        y_xm = 0.5_rt - axm;
-        y_xp = 0.5_rt - axp;
+        y_xm = 0.5_rt*dx[1] - axm;
+        y_xp = 0.5_rt*dx[1] - axp;
     }
 
-    barea(i,j,0) = nx*(axm-axp) + ny*(aym-ayp);
+    barea(i,j,0) = (nx*daxp + ny*dayp)/bareascaling;
     bcent(i,j,0,0) = 0.5_rt*(x_ym+x_yp);
     bcent(i,j,0,1) = 0.5_rt*(y_xm+y_xp);
     bnorm(i,j,0,0) = nx;
     bnorm(i,j,0,1) = ny;
 
     if (nxabs < tiny || nyabs > almostone) {
-        vfrac(i,j,0) = 0.5_rt*(axm+axp);
+        vfrac(i,j,0) = 0.5_rt*(axm+axp)*dx[0];
         vcent(i,j,0,0) = 0.0_rt;
-        if (vfrac(i,j,0) > almostone) {
+        if (vfrac(i,j,0)/dx[0] > almostone) {
```

Comment:
I think we should change this test to `vfrac(i,j,0)/(dx[0]*dx[1])`

Reply:
Thanks for spotting this, I fixed it (here and some lines below).